### PR TITLE
API: #10636, changing default of to_datetime to raise, deprecating coerce in favor of errors

### LINKED ISSUE #10636: API: change default in to_datetime for errors to 'raise'
xref #8894 
xref #10615 
xref #10154

The idea is that if you have completely unparseable dates with or w/o a format specified, then the default will be to raise. Further, since ``coerce`` is by default ``False``. You will get an immediate exception rather than have strings returned.

I think some tests will need to be adjusted and a simple example for the user in the whatsnew.

### PR DESCRIPTION
**Pull request by jreback**
closes #10636 

changes `to_timedelta` API as well to similar effect.

### CODE REVIEW

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -197,18 +197,24 @@ or ``format``, use ``to_datetime`` if these are required.
...
+
+Pass ``errors='coerce'`` to convert invalid data to ``NaT`` (not a time):
 
 .. ipython:: python
 
-   to_datetime(['2009-07-31', 'asd'])
+   to_datetime(['2009-07-31', 'asd'], errors='ignore')
 
-   to_datetime(['2009-07-31', 'asd'], coerce=True)
+   to_datetime(['2009-07-31', 'asd'], errors='coerce')
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: Can you add here the default of raising as well? The you see all three options next to each other

File: doc/source/whatsnew/v0.17.0.txt [hunk 2]
```diff
@@ -91,6 +92,38 @@ Backwards incompatible API changes
...
+     :okexcept:
+
+     pd.to_datetime(['2009-07-31', 'asd'])
+
+  Of course you can coerce this as well.
+
+  .. ipython:: python
+
+     to_datetime(['2009-07-31', 'asd'], errors='coerce')
+
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.17.0.txt]: Can you add something like:

```
To keep the previous behaviour, you can use `errors='ignore'`:

 .. ipython:: python

    to_datetime(['2009-07-31', 'asd'], errors='ignore')
```

File: pandas/tseries/timedeltas.py [hunk 3]
```diff
@@ -21,7 +24,10 @@ def to_timedelta(arg, unit='ns', box=True, coerce=False):
     box : boolean, default True
         If True returns a Timedelta/TimedeltaIndex of the results
         if False returns a np.timedelta64 or ndarray of values of dtype timedelta64[ns]
-    coerce : force errors to NaT (False by default)
+    errors : {'ignore', 'raise', 'coerce'}, default 'raise'
+        If 'raise', then invalid parsing will raise an exception
+        If 'coerce', then invalid parsing will be set as NaT
+        If 'ignore', then invalid parsing will return the input
```
Comment (jorisvandenbossche) [inline pandas/tseries/timedeltas.py]: can you add `-` markers for making this a list? (will be clearer in the html version)

Comment (jreback) [inline pandas/tseries/timedeltas.py]: done, though if you can check if this is ok

File: doc/source/whatsnew/v0.17.0.txt (hunk 2 above)
Comment (jreback) [inline doc/source/whatsnew/v0.17.0.txt]: done

File: doc/source/timeseries.rst (hunk 1 above)
Comment (jreback) [inline doc/source/timeseries.rst]: done

**Comment by jreback** [comment]
@jorisvandenbossche updated if you want to have a look again.

**Comment by jreback** [comment]
@jorisvandenbossche @shoyer @sinhrks @TomAugspurger any other comments?

**Comment by TomAugspurger** [comment]
Looks good.

**Comment by jorisvandenbossche** [comment]
Looks good!

One thing I was thinking: it would be nice if in the error message you get when there is a parsing error, that it also says something like "to coerce errors as NaN use "errors='coerce'" " if possible

**Comment by jreback** [comment]
@jorisvandenbossche your last point is nice, but I think a separate PR is in order. If you would like to open an issue gr8.

**Comment by jreback** [comment]
@jorisvandenbossche is there a way to limit tracebacks automatically in Sphinx (ones that are on purpose), other than using a code block

eg http://pandas-docs.github.io/pandas-docs-travis/whatsnew.html#changes-to-to-datetime-and-to-timedelta

**Comment by jorisvandenbossche** [comment]
no, I don't think so (in any case, the `okexcept` does not do that), but that is already a long time something I wanted to fix. Will try to do that for 0.17

And for the error message suggestion, certainly, that was no blocker for merging this!

**Pull request merged**